### PR TITLE
MISP2Sentinel

### PR DESCRIPTION
**Pull request by cudeso**
The MISP to Microsoft Sentinel integration allows you to upload indicators from MISP to Microsoft Sentinel.

New data connector for integration MISP to Sentinel
- JSON file with data connector definition
- Short README with link to GitHub repository where integration code is located

### CODE REVIEW

**Comment by cudeso** [comment]
@microsoft-github-policy-service agree

**Comment by v-rbajaj** [comment]
Hi @cudeso, thanks for contributing.
Are you done with the development? If yes, then please repackage the solution using this tool https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V3
Please go through the readme.md file and follow the instructions mentioned there.

Please do add release note as well, follow below mentioned link for the same.
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ReleaseNotesGuidance.md

**Comment by cudeso** [comment]
Hello @v-rbajaj thank you for the feedback. As far as I can verify, all checks have completed successfully.

**Comment by cudeso** [comment]
Please provide guidance on where these blanks are. I can't spot them in the templates.
```
Template Should Not Contain Blanks
#13 5.299     [-] Template Should Not Contain Blanks (180 ms)
#13 5.299         Empty property:  null Line: 229, Column: 35
```

**Comment by v-rbajaj** [comment]
> Please provide guidance on where these blanks are. I can't spot them in the templates.
> 
> ```
> Template Should Not Contain Blanks
> #13 5.299     [-] Template Should Not Contain Blanks (180 ms)
> #13 5.299         Empty property:  null Line: 229, Column: 35
> ```

Hi @cudeso, this is where the null value is 
image
When you update the maintemplate, please update the maintemplate inside 3.0.0 zip as well. 

One more thing, please update the "version" in release notes to 3.0.0 

We will review the rest of the contents.

**Comment by cudeso** [comment]
> > Please provide guidance on where these blanks are. I can't spot them in the templates.
> > ```
> > Template Should Not Contain Blanks
> > #13 5.299     [-] Template Should Not Contain Blanks (180 ms)
> > #13 5.299         Empty property:  null Line: 229, Column: 35
> > ```
> 
> Hi @cudeso, this is where the null value is image When you update the maintemplate, please update the maintemplate inside 3.0.0 zip as well.
> 
> One more thing, please update the "version" in release notes to 3.0.0
> 
> We will review the rest of the contents.

Hello. As far as I can check, that is in `mainTemplate.json`. That file is not created by me manually, it is the result of running the script `createSolutionV3.ps1` (from https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V3)

**Comment by v-rbajaj** [comment]
Hi @cudeso, we tried testing the maintemplate but there seems to be another issue apart from this, we will update you by 7 Aug 2023.

File: Solutions/MISP2Sentinel/SolutionMetadata.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"lastPublishDate": "2023-07-29",
+	"providers": ["MISP project", "cudeso.be"],
+	"categories": {
+		"domains" : ["Security - Threat Intelligence", "Security - Threat Protection"],
+		"verticals": ["Aeronautics", "Education", "Finance", "Healthcare", "Manufacturing", "Retail"]
+	},
+	"support": {
+	  "name": "MISP project & cudeso.be",
+	  "email": "info@cudeso.be",
+	  "tier": "Community-supported",
```
Comment (v-dvedak) [inline Solutions/MISP2Sentinel/SolutionMetadata.json:14]: The accepted tier values are Microsoft, Partner, Community. Please correct it. Thanks!

File: Solutions/MISP2Sentinel/SolutionMetadata.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+	"publisherId": "azuresentinel",
+	"offerId": "azure-sentinel-solution-misp2sentinel",
+	"firstPublishDate": "2023-07-29",
+	"lastPublishDate": "2023-07-29",
+	"providers": ["MISP project", "cudeso.be"],
```
Comment (v-dvedak) [inline Solutions/MISP2Sentinel/SolutionMetadata.json:6]: As per the standards we can have only one "providers". Please update accordingly.

File: Solutions/MISP2Sentinel/ReleaseNotes.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+| **Version** | **Date Modified (DD-MM-YYYY)** | **Change History**                          |
+|-------------|--------------------------------|---------------------------------------------|
+| 0.0.1       | 29-07-2023                     | **Data Connector** Initial version of MISP2Sentinel with support for Upload Indicators API
```
Comment (v-dvedak) [inline Solutions/MISP2Sentinel/ReleaseNotes.md:3]: Please check the version. As per me it should be 3.0.0.

**Comment by cudeso** [comment]
Hello @v-dvedak , do I need to re-run `createSolutionV3.ps1` (and first remove the `Package` folder) after updating `SolutionMetadata.json` and the release notes?

**Comment by v-rbajaj** [comment]
> Hello @v-dvedak , do I need to re-run `createSolutionV3.ps1` (and first remove the `Package` folder) after updating `SolutionMetadata.json` and the release notes?

Hi @cudeso, yes you need to re-run the createSolutionv3.ps1.

(Just fYI, we don't require to re run it for release notes, just for solutionmetadata.

**Comment by cudeso** [comment]
I did a pull of the repository to get any new versions of the tooling before running `createSolutionV3.ps1` again.
PR returned an error caused by `mainTemplate.json`, the file that was automatically created by createSolutionV3.ps1 ; similar behaviour as in https://github.com/Azure/Azure-Sentinel/pull/8658#issuecomment-1659905908

File: Solutions/MISP2Sentinel/ReleaseNotes.md (hunk 3 above)
Comment (cudeso) [reply Solutions/MISP2Sentinel/ReleaseNotes.md]: Changed in https://github.com/Azure/Azure-Sentinel/pull/8658/commits/01daea3dfc141fed17c97beae3d61d04e16749cb

File: Solutions/MISP2Sentinel/SolutionMetadata.json (hunk 2 above)
Comment (cudeso) [reply Solutions/MISP2Sentinel/SolutionMetadata.json]: changed in https://github.com/Azure/Azure-Sentinel/pull/8658/commits/01daea3dfc141fed17c97beae3d61d04e16749cb

File: Solutions/MISP2Sentinel/SolutionMetadata.json (hunk 1 above)
Comment (cudeso) [reply Solutions/MISP2Sentinel/SolutionMetadata.json]: changed in https://github.com/Azure/Azure-Sentinel/pull/8658/commits/01daea3dfc141fed17c97beae3d61d04e16749cb

**Comment by v-rbajaj** [comment]
Hi @cudeso, we are checking on the `"connectivityCriterias": null` because even after deploying the maintemplate we are getting this error. We will get back to you on 11 Aug 2023.
image

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**